Mesh_3/include/CGAL/Mesh_edge_criteria_3.h

```diff
@@ -121,58 +122,38 @@ class Mesh_edge_criteria_3
 
   /// \name Creation
   /// @{
+
   /*!
   * returns an object to serve as criteria for edges.
   *
   * \param length_bound is an upper bound
   * for the length of the edges which are used to discretize the curves.
+  * It can be a functional or a constant.
   * \param min_length_bound is a desired lower bound
   * for the length of the edges which are used to discretize the curves.
   * Only edges that are longer than this bound will be refined. Using
   * this lower bound can be handy on some domains, but using it may
   * break all the surface topology guarantees of the meshing algorithm.
   * It is not guaranteed to be exactly respected in the output mesh.
+  * \param distance_bound is an upper bound for the distance from the
+  * edge to the corresponding 1D feature.
+  * It can be a functional or a constant.
   *
   * \note If one parameter is set to 0, then its corresponding criterion is ignored.
-  */
-  Mesh_edge_criteria_3(const FT& length_bound,
-                       const FT& min_length_bound = 0)
-
-    : p_size_(new Mesh_3::internal::Sizing_field_container<
-                Mesh_constant_domain_field_3<GT,Index> ,
-                FT,
-                Point_3,
-                Index>(length_bound))
-    , min_length_bound_(min_length_bound)
-  {}
-
-  // Nb: SFINAE to avoid wrong matches with built-in numerical types
-  // as int.
-
-  /*!
-  * returns an object to serve as criteria for edges.
-  * The behavior and semantic of the argument are the same
-  * as above, except that the `length_bound`
-  * parameter is a functional instead of a constant.
   *
-  * @tparam SizingField a model of `MeshDomainField_3`
+  * @tparam SizingField and @tparam DistanceField are either scalars
+  * or models of `MeshDomainField_3`
   */
-  template < typename SizingField >
-  Mesh_edge_criteria_3
-  (
-
-   const SizingField& length_bound,
-   const FT& min_length_bound = 0
-#ifndef DOXYGEN_RUNNING
-    , std::enable_if_t<Mesh_3::Is_mesh_domain_field_3<Tr, SizingField>::value>* = 0
-#endif
-   )
-   : min_length_bound_(min_length_bound)
+  template < typename SizingField, typename DistanceField = FT >
+  Mesh_edge_criteria_3(const SizingField& length_bound,
+                       const FT& min_length_bound = 0,
+                       const DistanceField& distance_bound = FT(0))
```

Comment:
There are two parameters with default values. The constructor should use named parameters.

Reply:
the second one already is documented. Using named parameters would be a breaking change. Should we do it anyway?

Reply:
I agree, it might be a breaking change.

@sloriot I think you have tweaked CGAL named parameters for Mesh_3, to that it can stay backward-compatible, no?

Reply:
this is true for the previous NP `(facet_size = 1, tet_shape = 3)` converted to `facet_size(1).tet_shape(3)`, but `min_length_bound` has never been "named", so it would be a (maybe desirable) breaking change

Reply:
After discussion in person (IRL), we discovered that the back-compatibility I was thinking (two messages above) was not implemented. We discussed about it with @sloriot but there was technical details that really prevented to implement it. And so it is not availlable.

That means that if we decided to use named parameters here, that would break the compatibility. For CGAL-6.0, that would be OK, but it would be strange to add NP only to `Mesh_edge_criteria_3` and not for the other criteria classes for faces and cells.

So, eventually, we decided to let go. Let's keep the current API.